tgui/packages/tgui/interfaces/SupermatterMonitor.js

```diff
@@ -0,0 +1,147 @@
+import { Fragment } from 'inferno';
+import { useBackend } from '../backend';
+import { Section, Box, Button, Table, LabeledList, ProgressBar } from '../components';
+import { Window } from '../layouts';
+import { TableRow, TableCell } from '../components/Table';
+
+const powerToColor = power => {
+  if (power > 300) {
+    return 'bad';
+  } else if (power > 150) {
+    return 'average';
+  } else {
+    return 'good';
+  }
+};
+
+const temperatureToColor = temp => {
+  if (temp > 5000) {
+    return 'bad';
+  } else if (temp > 4000) {
+    return 'average';
+  } else {
+    return 'good';
+  }
+};
+
+const pressureToColor = pressure => {
+  if (pressure > 10000) {
+    return 'bad';
+  } else if (pressure > 5000) {
+    return 'average';
+  } else {
+    return 'good';
+  }
+};
+
+export const SupermatterMonitor = (props, context) => {
```

Comment:
I feel like the body of the main TGUI interface should be as compact as possible. You could separate the generated windows in the `if`/`else` into their own sub-interface (in the same file!) so that the main interface is as small as possible. Example:

```js
export const MyInterface = (props, context) => {
	const { act, data } = useBackend(context);
	if (data.screen === 1) {
		return <MyInterfaceScreenOne />;
	} else if (data.screen === 2) {
		return <MyInterfaceScreenTwo />;
	}

	// screen should be clamped server-side to prevent this location from being reached
}

// No export here so that this interface remains private (if it can be used somewhere else, it's better to put it in interfaces/common)
const MyInterfaceScreenOne = (props, context) => {
	const { act, data } = useBackend(context);
	// SCREEN ONE...
}

const MyInterfaceScreenTwo = (props, context) => {
	const { act, data } = useBackend(context);
	// SCREEN TWO...
}
```

Reply:
That method would have to make 3 callbacks back to BYOND (useBackend), whereas my current approach uses one. Im sticking with the method with the least overhead.